src/__tests__/store/ui/reducer.test.ts

```diff
@@ -0,0 +1,180 @@
+import { UiActionTypes, IUiState } from "../../../src/store/ui/types";
+import { uiReducer } from "../../../src/store/ui/reducer";
+
+const initialState: IUiState = {
+    loading: false,
+    progress: 0,
+    isDropdownOpen: false,
+    isKebabDropdownOpen: false,
+    isSidebarOpen: true,
+    sidebarActiveItem: "dashboard",
+    sidebarActiveGroup: "feeds_grp"
+};
+
+describe('ui reducer', () => {
+    it('should return the initial state', () => {
+        expect(uiReducer(undefined, {
+            type: null
+        })).toEqual(
+            initialState
+        )
+    })
+
+    it('should fetch request', () => {
+        const testState: IUiState = {
+            loading: true,
+            progress: 0,
+            isDropdownOpen: false,
+            isKebabDropdownOpen: false,
+            isSidebarOpen: true,
+            sidebarActiveItem: "dashboard",
+            sidebarActiveGroup: "feeds_grp"
+        };
+        expect(uiReducer(initialState, {
+            type: UiActionTypes.FETCH_REQUEST,
+            payload: testState.loading
+        })).toEqual(
+            {
+                loading: true,
+                progress: 0,
+                isDropdownOpen: false,
+                isKebabDropdownOpen: false,
+                isSidebarOpen: true,
+                sidebarActiveItem: "dashboard",
+                sidebarActiveGroup: "feeds_grp"
+            }
+        )
+    })
+
+    it('should fetch complete', () => {
+        const testState: IUiState = {
+            loading: false,
+            progress: 0,
+            isDropdownOpen: false,
+            isKebabDropdownOpen: false,
+            isSidebarOpen: true,
+            sidebarActiveItem: "dashboard",
+            sidebarActiveGroup: "feeds_grp"
+        };
+
+        expect(uiReducer(initialState, {
+            type: UiActionTypes.FETCH_COMPLETE,
+            payload: testState.loading
```

Comment:
this is good that you are testing the loading state, you just don't need the const above. just pass in the boolean payload